### PR TITLE
Don't remove app if test failed.

### PR DESCRIPTION
**Pull request by Katka92**
Don't remove app if the test fails. Applies to Basic Happy Path and Advanced Happy Path tests.
Related issue: [HAC-3982](https://issues.redhat.com/browse/HAC-3982)

### CODE REVIEW

File: pr_check.sh [hunk 1]
```diff
@@ -68,7 +68,8 @@ COMMON_SETUP="-v $WORKSPACE/artifacts:/tmp/artifacts:Z \
     -e CYPRESS_USERNAME=`echo ${B64_USER} | base64 -d` \
     -e CYPRESS_PASSWORD=`echo ${B64_PASS} | base64 -d` \
     -e CYPRESS_GH_PR_TITLE=${PR_TITLE} \
-    -e GH_COMMENTBODY=${GH_COMMENTBODY}"
+    -e GH_COMMENTBODY=${GH_COMMENTBODY} \
+    -e CYPRESS_REMOVE_APP_ON_FAIL=false"
```
Comment (jrichter1) [inline pr_check.sh:72]: I'd rather see this as a default value in the config file

File: integration-tests/tests/basic-happy-path.spec.ts [hunk 2]
```diff
@@ -19,7 +19,18 @@ describe('Basic Happy Path', { tags: ['@PR-check', '@publicRepo'] }, () => {
...
-    Applications.deleteApplication(applicationName);
+    // If some test failed, don't remove the app
+    let allTestsSucceeded = true;
+    if (
+      this.test.parent.eachTest((test) => {
+        if (test.state == 'failed') {
+          allTestsSucceeded = false;
+        }
+      })
+    )
```
Comment (jrichter1) [inline integration-tests/tests/basic-happy-path.spec.ts:30]: am I seeing this correctly, that the whole block is an if condition?

Comment (Katka92) [reply integration-tests/tests/basic-happy-path.spec.ts]: :woman_facepalming:

**Pull request merged**